_source/_posts/2022-05-02-angular-microfrontend-auth.md

```diff
@@ -0,0 +1,443 @@
+---
+layout: blog_post
+title: "How to build micro-frontends using Module Federation in Angular"
+author: alisa-duncan
+by: advocate
+communities: [javascript]
+description: "How to create an Angular web application with micro-frontends using Webpack 5 and Module Federation. In this first post of the series, you'll set up the project and share authenticated state in the web application."
+tags: [angular, typescript]
+tweets:
+- "Curious about micro-frontends and how to share state? 👀  Here's walkthrough on creating micro-frontends with #angular 🅰️"
+- "Build an @angular app using micro-frontends! This walkthrough uses #angular & Module Federation, shares state, and 🧁! 😋"
+image: 
+type: conversion
+github: https://github.com/oktadev/okta-angular-microfrontend-example
+---
+
+The demands placed on front-end web applications continue to grow. We expect our web applications to be feature-rich and highly performant as consumers. As developers, we worry about how to provide features and performance while keeping good development practices and architecture in mind.
+
+Enter micro-frontend architecture. Micro-frontends are modeled after the same concept as micro-services as a way to decompose monolithic front-ends. The micro-sized front-ends combine to form a fully-featured web app. Since each micro-frontend can be developed and deployed independently, you have a powerful way of scaling out front-end applications.
```

Comment:
```suggestion
Enter micro-frontend architecture. Micro frontends are modeled after the same concept as microservices, as a way to decompose monolithic frontends. You can combine micro-sized frontends  to form a fully-featured web app. Since each micro frontend can be developed and deployed independently, you have a powerful way of scaling out frontend applications.
```
Micro frontend - as a noun, you don't need the hyphen, use the hyphen for the adjective "micro-frontend architecture"